core/src/test/java/dk/alexandra/fresco/framework/network/AbstractCloseableNetworkTest.java

```diff
@@ -0,0 +1,352 @@
+package dk.alexandra.fresco.framework.network;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.fail;
+
+import dk.alexandra.fresco.framework.Party;
+import dk.alexandra.fresco.framework.configuration.NetworkConfiguration;
+import dk.alexandra.fresco.framework.configuration.NetworkConfigurationImpl;
+import dk.alexandra.fresco.framework.configuration.NetworkTestUtils;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import javax.net.ServerSocketFactory;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Abstracts tests for implementations of the {@link CloseableNetwork} interface.
+ *
+ */
+public abstract class AbstractCloseableNetworkTest {
```

Comment:
Yes! That is exactly how I wanted to do it. I really felt dirty using inheritance, but I couldn't find the right way to do it using JUnit ... do you have a pointer for this?

Reply:
The one way I could think of was to make `AbstractCloseableNetworkTest`  parameterized using the `@RunWith{Parameterized.class}` annotation stuff and then add a factory for each network implementation to the parameter list in `AbstractCloseableNetworkTest`. But that does not seem very elegant either, because `AbstractCloseableNetworkTest` would need to know about all the different implementations. Also, I would need separate test classes for each network implementation to test non-generic stuff. So in the end this seemed to be the cleaner and more modular. However, I would be super interested in alternatives.

Reply:
Maybe we should discuss this one offline, I am not following or adherering to your argument